inspirehep/modules/migrator/tasks.py

```diff
@@ -300,6 +300,12 @@ def create_record(record):
 
 def record_upsert(json):
     """Insert or update a record."""
+
+    def soft_delete_record():
+        if json.get('deleted'):
+            for pid in PersistentIdentifier.query.filter(PersistentIdentifier.object_uuid == record.id).all():
```

Comment:
No, it will happen that the same UUID have many PIDs. (e.g. recid, DOI, arXiv... or recid, ORCID, KAKEN...)